x-pack/platform/plugins/shared/streams_app/public/components/stream_list_view/documents_column.tsx

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiI18nNumber,
+  EuiSkeletonRectangle,
+  EuiDelayRender,
+} from '@elastic/eui';
+import { euiThemeVars } from '@kbn/ui-theme';
+import { css } from '@emotion/css';
+import moment from 'moment';
+import {
+  BarSeries,
+  Chart,
+  ScaleType,
+  Settings,
+  LIGHT_THEME,
+  DARK_THEME,
+  niceTimeFormatter,
+  Tooltip,
+  TooltipStickTo,
+  type SettingsProps,
+} from '@elastic/charts';
+import { i18n } from '@kbn/i18n';
+import { useEuiTheme } from '@elastic/eui';
+import { TimefilterHook } from '@kbn/data-plugin/public/query/timefilter/use_timefilter';
+import { useStreamsAppFetch } from '../../hooks/use_streams_app_fetch';
+import { useKibana } from '../../hooks/use_kibana';
+import { esqlResultToTimeseries } from '../../util/esql_result_to_timeseries';
+
+export function DocumentsColumn({
+  timefilter,
+  indexPattern,
+  numDataPoints,
+}: {
+  timefilter: TimefilterHook; // Workaround to keep state in sync
+  indexPattern: string;
+  numDataPoints: number;
+}) {
+  const {
+    dependencies: {
+      start: {
+        streams: { streamsRepositoryClient },
+      },
+    },
+  } = useKibana();
+
+  const { start, end } = timefilter.absoluteTimeRange;
+
+  const histogramQueryFetch = useStreamsAppFetch(
+    async ({ signal }) => {
+      const bucketSize = Math.round(moment.duration((end - start) / numDataPoints).asSeconds());
+      return streamsRepositoryClient.fetch('POST /internal/streams/esql', {
+        params: {
+          body: {
+            operationName: 'get_doc_count_for_stream',
+            query: `FROM ${indexPattern} | STATS doc_count = COUNT(*) BY @timestamp = BUCKET(@timestamp, ${bucketSize} seconds)`,
+            start,
+            end,
+          },
+        },
+        signal,
+      });
+    },
+    [streamsRepositoryClient, indexPattern, start, end, numDataPoints]
+  );
+
+  const allTimeseries = React.useMemo(
+    () =>
+      esqlResultToTimeseries({
+        result: histogramQueryFetch,
+        metricNames: ['doc_count'],
+      }),
+    [histogramQueryFetch]
+  );
+
+  const docCount = allTimeseries.reduce(
+    (acc, series) => acc + series.data.reduce((acc2, item) => acc2 + (item.doc_count || 0), 0),
```

Comment:
It doesn't matter for the classic streams, but there is a detail here we need to think about for wired streams - should we show the count for the stream and its children or just for the current layer? @LucaWintergerst any opinions here?